pageserver/src/tenant/timeline.rs

```diff
@@ -4155,6 +4161,169 @@ impl Timeline {
         false
     }
 
+    async fn create_image_layers_for_rel_blocks(
+        self: &Arc<Self>,
+        partition: &KeySpace,
+        mut image_layer_writer: ImageLayerWriter,
+        lsn: Lsn,
+        ctx: &RequestContext,
+        img_range: Range<Key>,
+        start: &mut Key,
+    ) -> Result<Vec<ResidentLayer>, CreateImageLayersError> {
```

Comment:
as discussed in the call: somewhere in this function we should also include an error (either logged or aborting compaction entirely) if the written image layer is larger than the target size. if we get to the upload limit, we should bail out of compaction entirely (until partitioning is implemented).

Reply:
I think it's fine to address Arpad's concern as a follow up since aux v2 is not enabled fleet wide.